Class not found when generating code coverage report

**Description:**
$subject
```
Exception in thread "main" java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/niveathika/WSO2/module-ballerina-java.jdbc/java.jdbc-ballerina/target/coverage/ballerina-java.jdbc-0.5.0-testable.jar/ballerina/java.jdbc
        at org.ballerinalang.test.runtime.util.CodeCoverageUtils.unzipCompiledSource(CodeCoverageUtils.java:76)
        at org.ballerinalang.test.runtime.entity.CoverageReport.generateReport(CoverageReport.java:78)
        at org.ballerinalang.test.runtime.CoverageMain.main(CoverageMain.java:47)
Caused by: java.nio.file.NoSuchFileException: /home/niveathika/WSO2/module-ballerina-java.jdbc/java.jdbc-ballerina/target/coverage/ballerina-java.jdbc-0.5.0-testable.jar/ballerina/java.jdbc
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:409)
        at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
        at java.nio.file.Files.move(Files.java:1395)
        at org.ballerinalang.test.runtime.util.CodeCoverageUtils.copyClassFilesToBinPath(CodeCoverageUtils.java:96)
        at org.ballerinalang.test.runtime.util.CodeCoverageUtils.unzipCompiledSource(CodeCoverageUtils.java:73)
        ... 2 more
```

The class seems to be generated under `/home/niveathika/WSO2/module-ballerina-java.jdbc/java.jdbc-ballerina/target/coverage/ballerina-java.jdbc-0.5.0-testable.jar/` and not under further directories of org & module as the error expects

--- a/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java
+++ b/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java
@@ -29,6 +29,7 @@ import org.jacoco.core.analysis.ISourceFileCoverage;
 import org.jacoco.core.tools.ExecFileLoader;
 
 import java.io.IOException;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
@@ -75,7 +76,12 @@ public class CoverageReport {
      * @throws IOException when file operations are failed
      */
     public void generateReport() throws IOException {
-        CodeCoverageUtils.unzipCompiledSource(sourceJarPath, projectDir, orgName, moduleName, version);
+        try {
+            CodeCoverageUtils.unzipCompiledSource(sourceJarPath, projectDir, orgName, moduleName, version);
+        } catch (NoSuchFileException e) {
+            return;
+        }
+
         execFileLoader.load(executionDataFile.toFile());
 
         final IBundleCoverage bundleCoverage = analyzeStructure();

--- a/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/CodeCoverageUtils.java
+++ b/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/CodeCoverageUtils.java
@@ -22,7 +22,9 @@ import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.PrintStream;
 import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
@@ -40,6 +42,8 @@ import java.util.zip.ZipInputStream;
  */
 public class CodeCoverageUtils {
 
+    private static final PrintStream errStream = System.err;
+
     /**
      * Util method to extract required class files for code coverage analysis.
      *
@@ -48,9 +52,10 @@ public class CodeCoverageUtils {
      * @param orgName org name of the project being executed
      * @param moduleName name of the module being executed
      * @param version version of the module being executed
+     * @throws NoSuchFileException if source file doesnt exist
      */
     public static void unzipCompiledSource(Path source, Path destination, String orgName, String moduleName,
-                                           String version) {
+                                           String version) throws NoSuchFileException {
         String destJarDir = destination.resolve(source.getFileName()).toString();
 
         try (JarFile jarFile = new JarFile(source.toFile())) {
@@ -70,8 +75,13 @@ public class CodeCoverageUtils {
                 }
 
             }
+
             copyClassFilesToBinPath(destination, destJarDir, orgName, moduleName, version);
             deleteDirectory(new File(destJarDir));
+        } catch (NoSuchFileException e) {
+            String msg = "Source file for " + moduleName + " doesnt exist. Cannot generate Code coverage";
+            errStream.println(msg);
+            throw new NoSuchFileException(msg);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }

--- a/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/CodeCoverageUtils.java
+++ b/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/CodeCoverageUtils.java
@@ -79,7 +79,8 @@ public class CodeCoverageUtils {
             copyClassFilesToBinPath(destination, destJarDir, orgName, moduleName, version);
             deleteDirectory(new File(destJarDir));
         } catch (NoSuchFileException e) {
-            String msg = "Source file for " + moduleName + " doesnt exist. Cannot generate Code coverage";
+            String msg = "Unable to generate code coverage for the module " + moduleName + ". Source file does not " +
+                    "exist";
             errStream.println(msg);
             throw new NoSuchFileException(msg);
         } catch (IOException e) {

